contracts/hooks/axelar/src/lib.rs

```diff
@@ -0,0 +1,292 @@
+            &mailbox,
+            &MailboxQueryMsg::LatestDispatchId {}.wrap(),
+        )
+        .or_else(|_| return Err(ContractError::LastDispatchQueryFailed {}));
+
+    let latest_dispatch_id = latest_dispatch_resp.unwrap().message_id;
+
+    let decoded_msg: Message = req.message.clone().into();
+
+    ensure_eq!(
+        latest_dispatch_id,
+        decoded_msg.id(),
+        ContractError::LastDispatchIDMismatch {
+            got: decoded_msg.id(),
+            expected: latest_dispatch_id
+        }
+    );
+
+    //send message to axelar gateway
+    let destination_chain = DESTINATION_CHAIN.load(deps.storage)?;
+    let destination_contract = DESTINATION_CONTRACT.load(deps.storage)?;
+    let destination_ism = DESTINATION_ISM.load(deps.storage)?;
+    let axelar_gateway_channel = AXELAR_GATEWAY_CHANNEL.load(deps.storage)?;
+
+    // TODO: do we need to pass a fee?
+    send_to_evm(
+        deps,
+        env,
+        info,
+        req.message,
+        axelar_gateway_channel,
+        destination_chain,
+        destination_contract,
+        vec![destination_ism],
+        None,
+    )
+}
+
+pub fn send_to_evm(
+    _deps: DepsMut,
+    env: Env,
+    _info: MessageInfo,
+    message: HexBinary,
+    gateway_channel: String,
+    destination_chain: String,
+    destination_contract: String,
+    destination_recipients: Vec<String>,
+    fee: Option<AxelarFee>,
+) -> Result<Response, ContractError> {
+    let addresses = destination_recipients
+        .into_iter()
+        .map(|s| match s.parse::<H160>() {
+            Ok(address) => Ok(Token::Address(Address::from(address))),
+            Err(_) => Err(ContractError::InvalidRecipientAddress { address: s }),
+        })
+        .collect::<Result<Vec<Token>, ContractError>>()?;
+
+    let payload = encode(&[Token::Array(addresses), Token::String(message.to_hex())]);
+
+    let msg = AxelarGeneralMessage {
+        destination_chain,
+        destination_address: destination_contract,
+        payload,
+        // TODO: extract constant
+        type_: 2,
+        fee,
+    };
+
+    // let coin = cw_utils::one_coin(&info).unwrap();
+    let ibc_transfer = MsgTransfer {
+        source_port: "transfer".to_string(),
+        source_channel: gateway_channel,
+        token: None, // TODO: is this gonna work?
+        sender: env.contract.address.to_string(),
+        receiver: AXELAR_GATEWAY.to_string(),
+        timeout_height: None,
+        timeout_timestamp: env.block.time.plus_seconds(604_800u64).nanos(),
+        memo: to_string(&msg).unwrap(),
+    };
+
+    // Base response
+    let response = Response::new()
+        .add_attribute("status", "ibc_message_created")
+        .add_attribute("ibc_message", format!("{:?}", ibc_transfer));
+
+    return Ok(response.add_message(ibc_transfer));
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> Result<QueryResponse, ContractError> {
```

Comment:
I feel that that for debugging and problem solving purposes this contract needs to store information for current hooks success/failure on dispatch. We can use sudo handlers for that. What do you guys think? @jcompagni10 @NicholasDotSol